rviz_default_plugins/src/rviz_default_plugins/ros_resource_retriever.cpp

```diff
@@ -115,7 +115,17 @@ RosResourceRetriever::get_shared(const std::string & url)
   req->path = url;
   req->etag = etag;
   auto result = this->client_->async_send_request(req);
-  executor_.spin_until_future_complete(result);
+
+  using namespace std::chrono_literals;
+  auto maximum_wait_time = 3s;
+
+  if (executor_.spin_until_future_complete(result, maximum_wait_time) !=
+    rclcpp::FutureReturnCode::SUCCESS)
+  {
+    RCLCPP_ERROR(this->logger_, "service call failed :(");
```

Comment:
Should this be an error or a warning? Should there be more information in the error, like for which asset it failed?